sdk/plugin/plugin.go

```diff
@@ -1,24 +1,36 @@
 package plugin
 
 import (
+	"context"
 	"io"
 	"log"
 	"os"
 	"sync"
 
 	"github.com/aserto-dev/ds-load/sdk/common/js"
-	"github.com/aserto-dev/ds-load/sdk/common/msg"
-	"github.com/aserto-dev/ds-load/sdk/transform"
-	"github.com/pkg/errors"
-	"google.golang.org/protobuf/encoding/protojson"
 )
 
+type Fetcher interface {
+	Fetch(ctx context.Context, outputWriter, errorWriter io.Writer) error
+}
+
+type Transform interface {
+	Transform(ctx context.Context, reader io.Reader, outputWriter, errorWriter io.Writer) error
+}
+
+type Publish interface {
```

Comment:
Please rename the interface to Publisher, following golang interface naming convention https://go.dev/doc/effective_go#interface-names